### PR TITLE
Legend in the viewport

### LINKED ISSUE #467: Sheet color legend with associated timing file
Render a legend in the viewport canvas to display which timing file a certain color belongs to.

### PR DESCRIPTION
**Pull request by Extremelyd1**
Fixes #467 

#### What have I done
Added a legend to the viewport to indicate what the different colors mean.

#### Testing
Automated testing.
Import files and check whether the legend correctly displays the colors and labels. Also check whether it is correct for multiple timing files.
Also try to toggle the legend in the configuration panel under the 'sheet' tab.

#### Addresses SRD section(s)
Viewport
Configuration Service

### CODE REVIEW

File: src/app/core/services/clear.service.ts [hunk 1]
```diff
@@ -66,8 +63,15 @@ export class ClearService {
...
 
+        // Clear the configuration to the default settings
+        this.configurationService.clear();
+
         // Clear the POI service
         this.poiService.clearSelectedPoiNames();
         this.poiService.clearPoiNames();
     }
+
+    public getConfigurationService() {
```
Comment (Dacaspex) [inline src/app/core/services/clear.service.ts]: Hallo?

File: src/app/workspace/viewport/utils/legend-entry.spec.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { LegendEntry } from './legend-entry';
+
```
Comment (Dacaspex) [inline src/app/workspace/viewport/utils/legend-entry.spec.ts]: Indentation

File: src/app/workspace/viewport/viewport-ui.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+     * The size of the colored boxes
+     */
+    private readonly COLOR_BOX_SIZE: number = 20;
+
+    /**
+     * Used for aligning the text with the box
+     */
+    private readonly ACADEMIC_HACK: number = 2;
+
+    constructor() {
```
Comment (Dacaspex) [inline src/app/workspace/viewport/viewport-ui.ts]: Empty constructor, can be gone

File: src/app/workspace/viewport/utils/legend-entry.spec.ts (hunk 2 above)
Comment (Extremelyd1) [reply src/app/workspace/viewport/utils/legend-entry.spec.ts]: Fixed

**Comment by Dacaspex** [comment]
Github did an OOPSIE

**Pull request closed without merging**